File: c/concepts/variables.c
#include <stdio.h>
#include <stdbool.h> // Required for bool, true, false
#include "concepts.h"

// Variables & Data Types
void demonstrateVariables() {
    printf("\n=== Variables and Data Types ===\n");

    // A string in C is an array of characters ending with a null terminator `\0`.
    char name[] = "John Doe";
    int age = 25;
    double height = 1.75;
    bool is_student = true;

    // C is statically typed. We state the known type in the output.
    printf("Name: %s (Type: char[])\n", name);
    printf("Age: %d (Type: int)\n", age);
    printf("Height: %f (Type: double)\n", height);
    printf("Is Student: %d (Type: bool)\n", is_student); // bools often print as 0 or 1
}

File: perl/concepts/variables.pl
# perl/concepts/variables.pl
use strict;
use warnings;

sub demonstrate_variables {
    print "\n=== Variables and Data Types ===\n";

    # Variables must be declared with `my`.
    # The `$` sigil denotes a scalar (single) value.
    my $name = "John Doe";
    my $age = 25;
    my $height = 1.75;
    # In Perl, true is any non-zero number or non-empty string.
    # False is 0, '0', '', or undef.
    my $is_student = 1; # Using 1 for true

    # String interpolation works inside double quotes.
    print "Name: $name (Type: Scalar)\n";
    print "Age: $age (Type: Scalar)\n";
    print "Height: $height (Type: Scalar)\n";
    print "Is Student: $is_student (Type: Scalar)\n";
}

# All files loaded via `require` must return a true value.
1;
